lib/Doctrine/DBAL/Tools/Console/ConsoleRunner.php

```diff
@@ -30,20 +36,30 @@ public static function createHelperSet(Connection $connection)
     }
 
     /**
-     * Runs console with the given helperset.
+     * Runs console with the given connection provider or helperset (deprecated).
      *
-     * @param Command[] $commands
+     * @param ConnectionProvider|HelperSet $helperSetOrConnectionProvider
+     * @param Command[]                    $commands
      *
      * @return void
      */
-    public static function run(HelperSet $helperSet, $commands = [])
+    public static function run($helperSetOrConnectionProvider, $commands = [])
     {
         $cli = new Application('Doctrine Command Line Interface', Version::VERSION);
 
         $cli->setCatchExceptions(true);
-        $cli->setHelperSet($helperSet);
 
-        self::addCommands($cli);
+        /** @var ConnectionProvider|null $connectionProvider */
+        $connectionProvider = $helperSetOrConnectionProvider;
```

Comment:
This assignment looks invalid. We know that the right-hand side value may be of the type `HelperSet` and yet we're assigning it to the variable named `$connectionProvider` and then set back to null in this case.

Please rework.